paperoni/db/database.py

```diff
@@ -493,6 +498,10 @@ def insert_flag(self, paper, flag_name, val):
         self.session.execute(ins_stmt)
         self.session.commit()
 
+        logger.info(
+            f"Paper {paper.title} ({paper.paper_id.hex()}) flagged {flag_name}={val}"
+        )
```

Comment:
I think it's a bit better to do this like that but I would prefer to use something like:

```python
curr = inspect.currentframe()
_args = inspect.getargvalues(curr)

# print method name with `curr.f_code.co_name` and args
```

But it seams like I can't make a `str()` in `paperoni.db.schema.Paper`

```python
def str(self):
    return f"{self.title} (self.paper_id.hex())"
```

`str(paper)` still prints something like `<paperoni.db.schema.Paper object at 0x11a3cf7d0>`

Reply:
Yeah, str() is not defined on these objects. I'm not sure what you are trying to do with that.

Reply:
I'd do something like that : 

```python
class Paper(Base):
    [...]

    def str(self):
        return f"{self.title} ({self.paper_id.hex()})"

class Database(OvldBase):
    [...]

    def insert_flag(self, paper, flag_name, val):
        [...]

        curr = inspect.currentframe()
        _args = inspect.getargvalues(curr)

        logger.info(
            curr.f_code.co_name + str({k:v for k, v, in _args if v is not self})
        )

    def insert_author_institution(
        self, author_id, institution_id, role, start_date, end_date
    ):
        [...]

        curr = inspect.currentframe()
        _args = inspect.getargvalues(curr)

        logger.info(
            curr.f_code.co_name + str({k:v for k, v, in _args if v is not self})
        )
```

but then `str(v)` would need to return something readable and not `<paperoni.db.schema.Paper object at 0x11a3cf7d0>`

Anyway, for now do you prefer the logs in the database methods or where the database calls are made (`validation.py`)?

Reply:
Not a bad idea, perhaps. For now this PR serves our needs well enough.